UltraStar Play/Assets/Common/Playlist/UltraStarPlaylistParser.cs

```diff
@@ -0,0 +1,126 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using UnityEngine;
+
+public static class UltraStarPlaylistParser
+{
+    public static readonly char separator = '-';
+
+    public static UltraStarPlaylist ParseFile(string path)
+    {
+        UltraStarPlaylist playlist = new UltraStarPlaylist();
+        string[] lines = File.ReadAllLines(path);
+        for (int lineIndex = 0; lineIndex < lines.Length; lineIndex++)
+        {
+            try
+            {
+                UltraStartPlaylistLineEntry lineEntry = UltraStarPlaylistLineParser.ParseLine(lines[lineIndex]);
+                playlist.AddLineEntry(lineEntry);
+            }
+            catch (Exception e)
+            {
+                Debug.LogException(new UltraStarPlaylistParserException($"Error in line {lineIndex + 1} in file '{path}'", e));
+            }
+        }
+        return playlist;
+    }
+
+    public static class UltraStarPlaylistLineParser
+    {
+        private enum Token
+        {
+            Artist, Title, Separator
+        }
+
+        public static UltraStartPlaylistLineEntry ParseLine(string line)
+        {
+            if (line.Trim().StartsWith("#"))
+            {
+                // This is a comment
+                return new UltraStartPlaylistLineEntry(line);
+            }
+
+            // Read the line character by character.
+            // Add this character either to the artist or title, depending on the targetToken.
+            Token targetToken = Token.Artist;
+            bool insideQuote = false;
+            string artist = "";
+            string title = "";
+            char lastChar = '0';
+            foreach (char c in line)
+            {
+                if (c == '"' && lastChar != '\\')
+                {
+                    insideQuote = !insideQuote;
+                    if (insideQuote)
+                    {
+                        // What has been read before for this token is now obsolete
+                        if (targetToken == Token.Artist)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this code to not nest more than 3 control flow statements.](https://app.codacy.com/manual/UltraStar-Deluxe/Play/pullRequest?prid=5574890)